src/main/java/commands/ScheduleCommand.java

```diff
@@ -1,11 +1,149 @@
+     */
+    public String getUserAction() {
+        return this.userAction;
+    }
+
+    /**
+     * Checks for the validity of the action specified by the user and sets the class field
+     * userAction to the action (if valid).
+     *
+     * @param userAction A string containing the action specified by the user.
+     * @throws InvalidCommandException If the action specified by the user is invalid.
+     */
+    public void setUserAction(String userAction) throws InvalidCommandException {
+        switch (userAction) {
+        case UPDATE_ACTION_KEYWORD:
+            //Fallthrough
+        case LIST_ACTION_KEYWORD:
+            //Fallthrough
+        case CLEAR_ACTION_KEYWORD:
+            //Fallthrough
+        case CLEAR_ALL_ACTION_KEYWORD:
+            this.userAction = userAction;
+            break;
+        default:
+            String className = this.getClass().getSimpleName();
+            throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
+        }
+    }
+
+    /**
+     * Gets the user argument stored in the class instance.
+     *
+     * @return A string containing the user argument.
+     */
+    public String getUserArguments() {
+        return this.userArguments;
+    }
+
+
+    @Override
+    public void execute() {
+        try {
+            switch (getUserAction()) {
+            case UPDATE_ACTION_KEYWORD:
+                Day newDay = scheduleList.updateDay(getUserArguments());
+                getUI().printNewScheduleCreatedMessage(newDay);
+                break;
+            case LIST_ACTION_KEYWORD:
+                break;
+            case CLEAR_ACTION_KEYWORD:
+                break;
+            case CLEAR_ALL_ACTION_KEYWORD: ;
+                break;
+            default:
+                String className = this.getClass().getSimpleName();
+                logger.log(Level.WARNING, "Invalid action under schedule command is entered!");
+                throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
+            }
+        } catch (InvalidCommandException e) {
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Uh oh, it seems like too few arguments were entered.");
+            System.out.println("Please try again. Alternatively, type 'help' if you need\n"
+                    + "more information on the commands.");
+        } catch (NumberFormatException e) {
+            logger.log(Level.WARNING, "A non-formattable number was received!");
+            System.out.println("Uh oh, a number was expected in your input, but a non-formattable\n"
+                    + "number was received.");
+            System.out.println("Please try again. Alternatively, type 'help' if you need\n"
+                    + "more information on the commands.");
```

Comment:
Perhaps can write the error messages to be printed as constants?

Reply:
Good suggestions will implement it